lib/utils/job-utils/http-tool.js

```diff
@@ -0,0 +1,202 @@
+//Copyright 2016, EMC, Inc.
+
+/*jshint node: true*/
+/*jshint multistr: true*/ 
+
+'use strict';
+
+var di = require('di');
+
+module.exports = HttpFactory;
+
+di.annotate(HttpFactory, new di.Provide('JobUtils.HttpTool'));
+di.annotate(HttpFactory, new di.Inject(
+    'Assert',
+    'Promise',
+    '_'
+));
+
+function HttpFactory(assert, Promise, _) {
+    
+    /**
+    * Tool class that does HTTP methods
+    * @constructor
+    */
+    function HttpTool(){
+        this.settings = {};
+        this.urlObject = {};
+        this.dataToWrite = '';
+        // ********** Helper Functions/Members **********
+        
+        this.validMethods = ['GET', 'PUT', 'POST', 'DELETE', 'PATCH'];
+        
+        /**
+        * Make sure that settings has at least property of url and method.
+        * @return {boolean} whether settings is valid.
+        */ 
+        this.isSettingValid = function(settings){
+            if (_.isEmpty(settings)) {
+                return false;
+            }
+            if ( ! ('url' in settings && 'method' in settings)) {
```

Comment:
Use `hasOwnProperty` here instead of `in`, below links explain why:
- http://stackoverflow.com/questions/13632999/if-key-in-object-or-ifobject-hasownpropertykey
- http://adripofjavascript.com/blog/drips/the-uses-of-in-vs-hasownproperty.html